src/Blazor.Diagrams.Core/Behaviors/DragMovablesBehavior.cs

```diff
@@ -5,12 +5,17 @@
 using Blazor.Diagrams.Core.Models.Base;
 using System;
 using System.Collections.Generic;
-
+using DiagramPoint = Blazor.Diagrams.Core.Geometry.Point;
 namespace Blazor.Diagrams.Core.Behaviors;
 
 public class DragMovablesBehavior : DragBehavior
 {
-    private readonly Dictionary<MovableModel, Point> _initialPositions;
+    public record NodeMoveablePositions(Point position)
+    {
+        public Dictionary<NodeModel, DiagramPoint> ChildPositions { get; } = new();
+    }
+
+    protected readonly Dictionary<NodeModel, NodeMoveablePositions> _initialPositions;
```

Comment:
I think we have some misunderstanding from yesterday. We should not add `ChildPositions` in Blazor.Diagrams, I thought we will just make `_initialPositions` protected so that we can reuse it in our implementation. Is there a solution where we use `_initialPositions` from this base class in our implementation? 
```suggestion
       protected readonly Dictionary<MovableModle, Point> _initialPositions;
```

Reply:
Updated